runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/config/HttpBindingConfig.java

```diff
@@ -164,6 +195,225 @@ private Function<Function<String, String>, String> asAccessor(
+                requestType.method = request.method;
+                requestType.contentType = request.contentType;
+                if (request.headers != null)
+                {
+                    requestType.headers = new HashMap<>();
+                    for (HttpParamConfig header : request.headers)
+                    {
+                        String8FW name = new String8FW(header.name);
+                        requestType.headers.put(name, createValidator.apply(header.validator, this.resolveId));
+                    }
+                }
+                if (request.pathParams != null)
+                {
+                    requestType.pathParams = new HashMap<>();
+                    for (HttpParamConfig pathParam : request.pathParams)
+                    {
+                        requestType.pathParams.put(pathParam.name, createValidator.apply(pathParam.validator, this.resolveId));
+                    }
+                }
+                if (request.queryParams != null)
+                {
+                    requestType.queryParams = new HashMap<>();
+                    for (HttpParamConfig queryParam : request.queryParams)
+                    {
+                        requestType.queryParams.put(queryParam.name, createValidator.apply(queryParam.validator, this.resolveId));
+                    }
+                }
+                if (request.content != null)
+                {
+                    requestType.content = createValidator.apply(request.content, this.resolveId);
+                }
+                result.add(requestType);
+            }
+        }
+        return result;
+    }
+
+    public HttpRequestType resolveRequest(
+        HttpBeginExFW beginEx)
+    {
+        HttpRequestType result = null;
+        if (requests != null && !requests.isEmpty())
+        {
+            String path = resolveHeaderValue(beginEx, HEADER_PATH);
+            String method = resolveHeaderValue(beginEx, HEADER_METHOD);
+            String contentType = resolveHeaderValue(beginEx, HEADER_CONTENT_TYPE);
+            for (HttpRequestType request : requests)
+            {
+                boolean isMatch = false;
+                isMatch |= method == null || request.method == null || method.equals(request.method.name());
+                isMatch |= contentType == null || request.contentType == null || request.contentType.contains(contentType);
+                isMatch &= parseParams(request, path);
+                if (isMatch)
+                {
+                    result = request;
+                    break;
+                }
+            }
+        }
+        return result;
+    }
+
+    private boolean parseParams(
+        HttpRequestType request,
+        String path)
+    {
+        boolean result = false;
+        String pattern = String.format("^%s/?(\\?(?<query0>.*))?$",
+            request.path.replaceAll("\\{([a-zA-Z_]+)\\}", "(?<$1>.+?)"));
+        Matcher matcher = Pattern.compile(pattern).matcher(path);
```

Comment:
This `pattern` and `matcher` should be precomputed once on `HttpRequestType` instead of recomputing on each call to `resolveRequest`.